action.yml

```diff
@@ -1,41 +1,67 @@
 name: 'Setup Rsync'
 description: 'Github actions to setup Rsync (all os supported) 🔄'
 
+inputs:
+  ssh_key:
+    description: ssh_key to setup
+    required: false
+
 runs:
   using: "composite"
   steps:
-    - name: Install Rsync
+    - name: Check OS
+      id: ostype
       run: |
         if [[ $OSTYPE == 'msys'* ]]; then
-          echo "windows runner detected"
-          curl.exe --output rsync-3.2.3-2-x86_64.pkg.tar.zst --url https://repo.msys2.org/msys/x86_64/rsync-3.2.3-1-x86_64.pkg.tar.zst
-          zstd -d rsync-3.2.3-2-x86_64.pkg.tar.zst
-          tar -xvf rsync-3.2.3-2-x86_64.pkg.tar
-          mv usr/bin/rsync.exe "C:\Program Files\Git\usr\bin\\"
-          curl.exe --output libzstd-1.5.2-1-x86_64.pkg.tar.zst https://repo.msys2.org/msys/x86_64/libzstd-1.5.2-1-x86_64.pkg.tar.zst
-          zstd -d libzstd-1.5.2-1-x86_64.pkg.tar.zst
-          tar -xvf libzstd-1.5.2-1-x86_64.pkg.tar
-          mv usr/bin/msys-zstd-1.dll "C:\Program Files\Git\usr\bin\\"
-          curl.exe --output libxxhash-0.8.0-1-x86_64.pkg.tar.zst https://repo.msys2.org/msys/x86_64/libxxhash-0.8.0-1-x86_64.pkg.tar.zst
-          zstd -d libxxhash-0.8.0-1-x86_64.pkg.tar.zst
-          tar -xvf libxxhash-0.8.0-1-x86_64.pkg.tar
-          mv usr/bin/msys-xxhash-0.8.0.dll "C:\Program Files\Git\usr\bin\\"
-          echo "rsync installed on windows runner"
+          echo "Windows detected"
+          echo "ostype=windows" >> $GITHUB_OUTPUT
         elif [[ $OSTYPE == 'darwin'* ]]; then 
-          echo "macos runner detected"
-          brew install rsync
-          echo "rsync installed on macos runner"
+          echo "MacOS detected"
+          echo "ostype=macos" >> $GITHUB_OUTPUT
         else
-          echo "ubuntu runner detected"
-          sudo apt-get install rsync
-          echo "rsync installed on ubuntu runner"
+          echo "Linux detected"
+          echo "ostype=linux" >> $GITHUB_OUTPUT
         fi
       shell: bash
 
+    - name: Install Rsync Windows
+      if: steps.ostype.outputs.ostype == 'windows'
+      run: |
+        choco install --no-progress rsync
+      shell: bash
+    
+    - name: Install Rsync MacOS
+      if: steps.ostype.outputs.ostype == 'macos'
+      run: |
+        echo "macos runner detected"
+        brew install rsync
+        echo "rsync installed on macos runner"
+      shell: bash
+    
+    - name: Install Rsync Ubuntu
+      if: steps.ostype.outputs.ostype == 'linux'
+      run: |
+        echo "ubuntu runner detected"
+        sudo apt-get install rsync
+        echo "rsync installed on ubuntu runner"
```

Comment:
This always outputs that rsync is installed. Maybe, skip installation - and if GitHub updates the image - and removes rsync, this action can be updated.

Reply:
Does one need `apt-get update` before? - On "normal" docker images, this is required, because the indexes are normally wiped away to save space.

Some bookmarks to share:

- Run GitHub actions locally: https://github.com/nektos/act 
  - (rather small) Image with sudo: https://github.com/fwilhe2/act-runner-image 
  - act --rm --platform ubuntu-latest=fwilhe2/act-runner:latest -W Tests

Reply:
I've update this section in the action.yaml file if you want to have a look 👍🏼